pkg/spec3/parameter.go

```diff
@@ -51,6 +54,18 @@ func (p *Parameter) MarshalJSON() ([]byte, error) {
 	return swag.ConcatJSON(b1, b2, b3), nil
 }
 
+func (p *Parameter) MarshalNextJSON(opts jsonv2.MarshalOptions, enc *jsonv2.Encoder) error {
+	var x struct {
+		Ref            string                 `json:"$ref,omitempty"`
+		ParameterProps parameterPropsOmitZero `json:",inline"`
+		spec.Extensions
+	}
+	x.Ref = p.Refable.Ref.String()
+	x.Extensions = internal.SanitizeExtensions(p.Extensions)
+	x.ParameterProps = parameterPropsOmitZero(p.ParameterProps)
```

Comment:
Could changing these `parameterPropsOmitZero`, `operationsPropsOmitZero`, etc. types to a pointer avoid a copy and boost performance at all?

Reply:
I tried something like

```
    var x struct {
        Ref            string                 `json:"$ref,omitempty"`
+       ParameterProps *parameterPropsOmitZero `json:",inline"`
        spec.Extensions
    }
    ...
+   tmp := parameterPropsOmitZero(p.ParameterProps)
+   x.ParameterProps = &tmp
    ...
```

and performance numbers look identical. Is this what you mean by changing them to pointers?

Reply:
Yeah, Jordan called out the same thing in my PR for V2, and I ran the same experiment :-)

Reply:
I meant using like this:

`x.ParameterProps = (*parameterPropsOmitZero)(&p.ParameterProps)`

Reply:
https://github.com/kubernetes/kube-openapi/pull/325#discussion_r1028573872

Reply:
```
| Name                                               | Before       | After        |
| (appsv1)OpenAPIV3Deserialize/jsonv2-24             | 13.68m ± 1%  | 13.65m ± 1%  |
| (authorizationv1)OpenAPIV3Deserialize/jsonv2#01-24 | 1.199m ± 1%  | 1.192m ± 2%  |
| (appsv1)OpenAPIV3Deserialize/jsonv2-24             | 5.260Mi ± 0% | 5.260Mi ± 0% |
| (authorizationv1)OpenAPIV3Deserialize/jsonv2#01-24 | 525.4Ki ± 0% | 525.4Ki ± 0% |
| (appsv1)OpenAPIV3Deserialize/jsonv2-24             | 26.89k ± 0%  | 26.89k ± 0%  |
| (authorizationv1)OpenAPIV3Deserialize/jsonv2#01-24 | 2.517k ± 0%  | 2.517k ± 0%  |
```

Reply:
Yeah doesn’t seem to do anything